src/main/scala/mesosphere/marathon/core/CoreModuleImpl.scala

```diff
@@ -53,7 +55,8 @@ class CoreModuleImpl @Inject() (
   scheduler: Provider[DeploymentService],
   instanceUpdateSteps: Seq[InstanceChangeHandler],
   taskStatusUpdateProcessor: TaskStatusUpdateProcessor,
-  homeRegionProvider: Provider[HomeRegionProvider]
+  mesosLeaderInfo: MesosLeaderInfo,
+  @Named(ModuleNames.MESOS_HEARTBEAT_ACTOR) heartbeatActor: ActorRef
```

Comment:
Why was `@Named(ModuleNames.MESOS_HEARTBEAT_ACTOR) heartbeatActor: ActorRef` added?

Reply:
that is a dependency of MesosHeartbeatMonitor which I instantiated here instead of by guice. Also MarathonSchedulerService depends on it so I kept it managed by guice for now...

Reply:
Ah, too much Guice magic for me ^^.